webapp/pages/settings/my-social-media.spec.js

```diff
@@ -43,79 +38,152 @@ describe('my-social-media.vue', () => {
   })
 
   describe('mount', () => {
+    let form, input, submitButton
     const Wrapper = () => {
-      store = new Vuex.Store({
+      const store = new Vuex.Store({
         getters,
       })
       return mount(MySocialMedia, { store, mocks, localVue })
     }
 
-    it('renders', () => {
-      wrapper = Wrapper()
-      expect(wrapper.contains('div')).toBe(true)
+    describe('adding social media link', () => {
+      beforeEach(() => {
+        wrapper = Wrapper()
+        form = wrapper.find('form')
+        input = wrapper.find('input#addSocialMedia')
+        submitButton = wrapper.find('button')
+      })
+
+      it('requires the link to be a valid url', () => {
+        input.setValue('some value')
+        form.trigger('submit')
+
+        expect(mocks.$apollo.mutate).not.toHaveBeenCalled()
+      })
+
+      it('displays an error message when not saved successfully', async () => {
+        mocks.$apollo.mutate.mockRejectedValue({ message: 'Ouch!' })
+        input.setValue(newSocialMediaUrl)
+        form.trigger('submit')
+
+        await flushPromises()
+
+        expect(mocks.$toast.error).toHaveBeenCalledTimes(1)
+      })
+
+      describe('success', () => {
+        beforeEach(() => {
+          mocks.$apollo.mutate.mockResolvedValue({
+            data: { CreateSocialMeda: { id: 's2', url: newSocialMediaUrl } },
+          })
+          input.setValue(newSocialMediaUrl)
+          form.trigger('submit')
+        })
+
+        it('sends the new url to the backend', () => {
+          const expected = expect.objectContaining({
+            variables: { url: newSocialMediaUrl },
+          })
+
+          expect(mocks.$apollo.mutate).toHaveBeenCalledTimes(1)
+          expect(mocks.$apollo.mutate).toHaveBeenCalledWith(expected)
```

Comment:
it's best to avoid multiple expectations in the same it block, I would be happy with the `toHaveBeenCalledWith(expected)`
In my opinion that is a strong test, but maybe you had your motives for adding that it was also only called once?